src/tribler-core/tribler_core/modules/popularity/popularity_community.py

```diff
@@ -29,36 +38,77 @@ def __init__(self, *args, **kwargs):
 
         self.add_message_handler(TorrentsHealthPayload, self.on_torrents_health)
 
-        self.logger.info('Popularity Community initialized (peer mid %s)', hexlify(self.my_peer.mid))
-        self.register_task("publish", self.gossip_torrents_health, interval=PUBLISH_INTERVAL)
+        self.logger.info('Popularity Community initialized (peer mid %s)',
+                         hexlify(self.my_peer.mid))
+        self.register_task("gossip", self.gossip_torrents_health,
+                           interval=PopularityCommunity.GOSSIP_INTERVAL)
+
+    @staticmethod
+    def select_torrents_to_gossip(torrents):
+        """ Select torrents to gossip.
+
+        Select top 5 popular torrents, and 5 random torrents.
+
+        Args:
+            torrents: set of tuples (infohash, seeders, leechers, last_check)
+
+        Returns:
+            tuple (set(popular), set(random))
+
+        """
+        # select the torrents that have seeders
+        alive = set((_, seeders, *rest) for (_, seeders, *rest) in torrents
```

Comment:
I think here, if the `alive` set is empty then simply two empty set or None could be returned.

Reply:
Added